### DISCUSSION TITLE
Help to understand why solidjs hydrate doesn't change DOM

### DISCUSSION THREAD

**Discussion by vchirikov** [vikejs/vike#851] [Discussion] [answered]
Hi,

I'm trying to use solidjs + vps and found strange hydration issue, I don't get is it my code / vps / solidjs issue, so I created a minimal repro from example/solid-ssr-v1/ take a look:

https://github.com/vchirikov/solidjs-repro

![issue](https://github.com/vchirikov/solidjs-repro/raw/master/issue.gif)

Could you help to understand why `hydrate()` doesn't change value after server?
I mean we see correct value in the console, but not in the DOM. if we use `render()` instead of hydrate we will see the correct value in the added dom to `#page-view`

![image](https://user-images.githubusercontent.com/40987940/235871412-274598b0-13a8-4c52-93fd-fe9514eed764.png)

**Comment by vchirikov** [comment]
https://github.com/vchirikov/solidjs-repro/compare/master...workaround-01#diff-281637db84af8f02853378a189c2c835fb01180e281e67530f422c85d5ecc311

This is a hacky way to force refresh container after hydration, but looks like I miss something to understand the problem :\

**Reply by vchirikov** [reply]
but this doesn't work with inversifyjs container...ohh

**Comment by vchirikov** [comment]
@ryansolid could you help to understand how to provide `container` to the hydrate to correct rerendering all the parts? or at least check that there is no bugs in hydrate() related to this.

**Comment by brillout** [comment]
FYI some reference implementations:
 - https://github.com/magne4000/solide
 - [/examples/solid-ssr-v1/](https://github.com/brillout/vite-plugin-ssr/tree/main/examples/solid-ssr-v1)

**Reply by vchirikov** [reply]
repro already based on your [/examples/solid-ssr-v1/](https://github.com/brillout/vite-plugin-ssr/tree/main/examples/solid-ssr-v1), the problem is hydrate() doesn't change the DOM, but I don't know the way to force rehydration all the page, 
I will take a look at solide anyway thanks for the reply

**Reply by brillout** [reply]
Seems to me that you're trying to force-apply React's mindset onto Solid, but Solid's reactivity works completely differently.

**Reply by vchirikov** [reply]
I want to use 2 IoC containers (inversify-js): one for server-side render and one container for client-side, for example - I want to render page ssg (ssr) with `UserService` which return `Anonymous` on server-side and another implementation from client-side container read username from localstorage and `hydrate()` should change DOM, but for some reason it doesn't, although code in e.g. `createMemo()` returns correct values, but hydrate() doesn't see the diff :(

**Comment by vchirikov** [comment, marked as answer]
I'm pretty sure that this is a solidjs bug, not related to vps, so I've created issue there https://github.com/solidjs/solid/issues/1716 , going to close this discussion, because it's not related to vite-plugin-ssr. Thanks

**Reply by brillout** [reply]
Thanks for circling back on this.
